app/src/main/java/org/schabi/newpipe/notifications/ScheduleOptions.kt

```diff
@@ -0,0 +1,31 @@
+package org.schabi.newpipe.notifications
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+import org.schabi.newpipe.R
+import java.util.concurrent.TimeUnit
+
+data class ScheduleOptions(
+    val interval: Long,
+    val isRequireNonMeteredNetwork: Boolean
+) {
+
+    companion object {
+
+        fun from(context: Context): ScheduleOptions {
+            val preferences = PreferenceManager.getDefaultSharedPreferences(context)
+            return ScheduleOptions(
+                interval = TimeUnit.HOURS.toMillis(
+                    preferences.getString(
+                        context.getString(R.string.streams_notifications_interval_key),
+                        context.getString(R.string.streams_notifications_interval_default)
+                    )?.toLongOrNull() ?: 6
```

Comment:
Extract this `6` into a constant. Also, isn't it strage that you are using `6` hours as the default, when `6` is not even an option in settings? Just use `3`, i.e. the same as `streams_notifications_interval_default`.

Reply:
\+